models/lowpassfilter_spike_detector.cpp

```diff
@@ -0,0 +1,863 @@
+{
+}
+
+nest::lowpassfilter_spike_detector::Parameters_::Parameters_(
+  const Parameters_& p )
+  : record_spikes_( p.record_spikes_ )
+  , tau_( p.tau_ )
+  , filter_start_times_( p.filter_start_times_ )
+  , filter_stop_times_( p.filter_stop_times_ )
+  , filter_report_interval_( p.filter_report_interval_ )
+{
+  filter_report_interval_.calibrate();
+}
+
+nest::lowpassfilter_spike_detector::State_::State_()
+{
+}
+
+void
+nest::lowpassfilter_spike_detector::Parameters_::get( DictionaryDatum& d ) const
+{
+  ( *d )[ names::filter_start_times ] = filter_start_times_;
+  ( *d )[ names::filter_stop_times ] = filter_stop_times_;
+  ( *d )[ names::filter_report_interval ] = filter_report_interval_.get_ms();
+  ( *d )[ names::tau_filter ] = tau_;
+  ( *d )[ names::record_spikes ] = record_spikes_;
+}
+
+nest::lowpassfilter_spike_detector::Variables_::Variables_()
+  : filter_block_index_( 0 )
+{
+}
+
+void
+nest::lowpassfilter_spike_detector::init_state_( const Node& np )
+{
+  const lowpassfilter_spike_detector& sd =
+    dynamic_cast< const lowpassfilter_spike_detector& >( np );
+  spikes_device_.init_state( sd.spikes_device_ );
+  filtered_device_.init_state( sd.filtered_device_ );
+
+  init_buffers_();
+}
+
+void
+nest::lowpassfilter_spike_detector::init_buffers_()
+{
+  spikes_device_.init_buffers();
+  filtered_device_.init_buffers();
+
+  std::vector< double > tmp_filter_times( 0 );
+  S_.filter_times_.swap( tmp_filter_times );
+
+  std::vector< std::vector< double > > tmp_node_traces(
+    S_.node_gids_.size(), std::vector< double >() );
+  S_.node_traces_.swap( tmp_node_traces );
+
+  // Size is known.
+  std::vector< double > tmp_trace_times( S_.node_gids_.size(), 0.0 );
+  B_.trace_times_.swap( tmp_trace_times );
+
+  // Size is known.
+  std::vector< double > tmp_traces( S_.node_gids_.size(), 0.0 );
+  B_.traces_.swap( tmp_traces );
+
+  std::vector< long > tmp_steps_to_filter( 0 );
+  B_.steps_to_filter_.swap( tmp_steps_to_filter );
+
+  std::vector< std::vector< std::vector< Event* > > > tmp_node_spikes(
+    2,
+    std::vector< std::vector< Event* > >(
+      S_.node_gids_.size(), std::vector< Event* >() ) );
+  B_.node_spikes_.swap( tmp_node_spikes );
+}
+
+void
+nest::lowpassfilter_spike_detector::calibrate()
+{
+  if ( kernel().event_delivery_manager.get_off_grid_communication()
+    and not spikes_device_.is_precise_times_user_set() )
+  {
+    spikes_device_.set_precise_times( true );
+    filtered_device_.set_precise_times( true );
+    std::string msg = String::compose(
+      "Precise neuron models exist: the property precise_times "
+      "of the %1 with gid %2 has been set to true",
+      get_name(),
+      get_gid() );
+
+    if ( spikes_device_.is_precision_user_set() )
+    {
+      // if user explicitly set the precision, there is no need to do anything.
+      msg += ".";
+    }
+
+    else
+    {
+      // it makes sense to increase the precision if precise models are used.
```

Comment:
Yes it is! I already took care of it in c7ec080db108120165f63c22e7f81079a03a418b and 6f9c841a119e7c3075b8a1bfcee5a6ad04a8a24b .